llvm/include/llvm/ADT/APFloat.h

```diff
@@ -1091,6 +1103,31 @@ class APFloat : public APFloatBase {
     }
   }
 
+  /// Returns true if the given semantics can represent Zero.
+  ///
+  /// \param Sem - type float semantics
+  static bool hasZero(const fltSemantics &Sem) {
+    switch (SemanticsToEnum(Sem)) {
```

Comment:
Just directly check &Sem == the one? There's only one case so the switch isn't helping 

I find the whole SemanticsToEnum and opaque fltSemantics queries really obnoxious. I don't see why fltSemantics doesn't just have public fields

Reply:
Yes, that makes sense for now, since there is only one type.
Fixed it in the latest revision.